clustered/server/src/main/java/org/ehcache/clustered/server/ServerSideServerStore.java

```diff
@@ -27,7 +27,8 @@
 
 @CommonComponent
 public interface ServerSideServerStore extends ServerStore, MapInternals {
-  void setEvictionListener(ServerStoreEvictionListener listener);
+  void setEventListener(ServerStoreEventListener listener);
+  void enableEvents(boolean enable);
```

Comment:
There unfortunately are two places where these events have a non-trivial extra cost:

- The `OffHeapServerStore.append()` call at the very bottom of the chain needs to perform a `getAndAppend` instead of a simple `append` when events are enabled.
- Eviction callback has to copy the evicted chain into the heap and encode it as part of the client invalidation notification.